From 64a2045936ffb61f953f1c8c5ffcd9294f7e650e Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Wed, 11 Nov 2015 09:31:04 +0800
Subject: [PATCH 1255/1691] MLK-11851 tty: series: imx: add dma memory check
 before freeing

commit 64a2045936ffb61f953f1c8c5ffcd9294f7e650e from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

Add dma memory check before free it.

Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 drivers/tty/serial/imx.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index 25c0030..5362259 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -981,10 +981,12 @@ static void imx_uart_dma_exit(struct imx_port *sport)
 		dma_release_channel(sport->dma_chan_rx);
 		sport->dma_chan_rx = NULL;
 
-		dma_free_coherent(NULL, IMX_RXBD_NUM * RX_BUF_SIZE,
-					(void *)sport->rx_buf.buf,
-					sport->rx_buf.dmaaddr);
-		sport->rx_buf.buf = NULL;
+		if (sport->rx_buf.buf) {
+			dma_free_coherent(NULL, IMX_RXBD_NUM * RX_BUF_SIZE,
+						(void *)sport->rx_buf.buf,
+						sport->rx_buf.dmaaddr);
+			sport->rx_buf.buf = NULL;
+		}
 	}
 
 	if (sport->dma_chan_tx) {
-- 
1.9.1

